plugins/plugins.js

```diff
@@ -0,0 +1,172 @@
+      return function (pluginName, cb) {
+        chechInstalled(pluginName, function (err) {
+          if (err) return cb(err)
+
+          config.plugins[pluginName] = b 
+          writePluginConfig(pluginName, b)
+          if (b)
+            cb(null, '\''+pluginName+'\' has been enabled. Restart Scuttlebot server to use the plugin.')
+          else
+            cb(null, '\''+pluginName+'\' has been disabled. Restart Scuttlebot server to stop using the plugin.')
+        })
+      }
+    }
+
+    // helper to check if a plugin is installed
+    function chechInstalled (pluginName, cb) {
+      if (!pluginName || typeof pluginName !== 'string')
+        return cb(new Error('plugin name is required'))
+      var modulePath = path.join(installPath, 'node_modules', pluginName)
+      fs.stat(modulePath, function (err) {
+        if (err)
+          cb(new Error('Plugin "'+pluginName+'" is not installed.'))
+        else
+          cb()
+      })
+    }
+
+    // write the plugin config to ~/.ssb/config
+    function writePluginConfig (pluginName, value) {
+      var cfgPath = path.join(config.path, 'config')
+      var existingConfig = {}
+      
+      // load ~/.ssb/config
+      try { existingConfig = JSON.parse(fs.readFileSync(cfgPath, 'utf-8')) }
+      catch (e) {}
+
+      // update the plugins config
+      existingConfig.plugins = existingConfig.plugins || {}
+      existingConfig.plugins[pluginName] = value
+
+      // write to disc
+      fs.writeFileSync(cfgPath, JSON.stringify(existingConfig, null, 2), 'utf-8')
+    }
+
+    return { 
+      install: function (pluginName, opts) {
+        var p = pushable()
+        var dryRun = opts && opts['dry-run']
+
+        if (!pluginName)
+          return pull.error(new Error('plugin name is required'))
+
+        // build args
+        // --global-style: dont dedup at the top level, gives proper isolation between each plugin
+        // --loglevel error: dont output warnings, because npm just whines about the lack of a package.json in ~/.ssb
+        var args = ['install', pluginName, '--global-style', '--loglevel', 'error']
+        if (dryRun)
+          args.push('--dry-run')
+
+        // exec npm
+        var child = spawn('npm', args, { cwd: installPath })
+          .on('close', function (code) {
+            if (code == 0 && !dryRun)
+              p.push(new Buffer('"'+pluginName+'" has been installed. Restart Scuttlebot server to enable the plugin.\n', 'utf-8'))
+
+            // enable the plugin
+            // - use basename(), because plugins can be installed from the FS, in which case pluginName is a path
+            var name = path.basename(pluginName)
+            config.plugins[name] = true
+            writePluginConfig(name, true)
+            p.end()
+          })
+        return cat([
+          pull.values([new Buffer('Installing "'+pluginName+'"...\n', 'utf-8')]),
+          toPull(child.stdout),
+          toPull(child.stderr),
```

Comment:
instead of concatenating the stdout and stderr it would be better to use pull-many so they come out the same as you'd normally see it.

`cat([installing, many(stdout, stderr), p])`